tsl/src/continuous_aggs/refresh.c

```diff
@@ -475,15 +473,20 @@ continuous_agg_refresh_internal(const ContinuousAgg *cagg,
 	 * prevent transaction blocks.  */
 	PreventInTransactionBlock(true, REFRESH_FUNCTION_NAME);
 
-	if (refresh_window_arg->start >= refresh_window_arg->end)
+	refresh_window =
+		compute_inscribed_bucketed_refresh_window(refresh_window_arg, cagg->data.bucket_width);
+
+	if (refresh_window.start >= refresh_window.end)
 		ereport(ERROR,
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("invalid refresh window"),
-				 errhint("The start of the window must be before the end.")));
+				 errmsg("refresh window too small"),
+				 errdetail("The refresh window must enclose at least one bucket of data."),
```

Comment:
Suggest to add information about the start and end of the window here since that can help debugging problems.